### ISSUE TITLE
[Bug]: wrong unit conversion in mean climate diagnostics 

### ISSUE THREAD

**Issue by zhangshixuan1987** [PCMDI/pcmdi_metrics#1128]
### What happened?

The library of mean climate diagnostics includes a "load_and_regrid.py" module which attempts to extract the test and reference data at the pressure level. In the following lines:

- https://github.com/PCMDI/pcmdi_metrics/blob/fd176464087460656fcd65d2bfce3997f7e90c45/pcmdi_metrics/mean_climate/lib/load_and_regrid.py#L95C1-L98C49,
- 
below code block was added: 
```
if "plev" in list(ds.coords.keys()):
            if ds.plev.units == "Pa":
                level = level * 100  # hPa to Pa
            try:
                ds = ds.sel(plev=level)
```

While in the ``mean_climate_driver.py" module, following
[code block](https://github.com/PCMDI/pcmdi_metrics/blob/fd176464087460656fcd65d2bfce3997f7e90c45/pcmdi_metrics/mean_climate/mean_climate_driver.py#L167-L170) show up: 

```
    result_dict["Variable"] = dict()
    result_dict["Variable"]["id"] = varname
    if level is not None:
        result_dict["Variable"]["level"] = level * 100  # hPa to Pa
```

This seems to be problematic, as you have converted the unit of **level** from hPa to Pa in **mean_climate_driver.py**, and then in the **load_and_regrid.py**, **level** times 100 again when the unit of **plev** in data file is "Pa".  Usually, the unit convention for **plev** should be "Pa",  although the unit of **plev** can be "hPa" or "mb" sometimes (depend on how user prepare the data).   

Overall, I think that the unit conversion in the **mean_climate_driver.py** should be removed, and it is better to let the **load_and_regrid.py** deal with the unit conversion properly. 

 

### Environment

I am using the PCMDI master branch and run it on Argon Chryslis machine.

**Comment by lee1043** [comment]
Hi @zhangshixuan1987, thank you for the post. I see your point that it could be problematic particularly when the plev in the data is not [Pa]. As PMP has been developed with multi-model context, [Pa] for plev has been expected for the most of time as a result of CMOR. 

The current code works as follows:
1. in the parameter file level is expected to be given as hPa (e.g., 850)
2. level [hPa] info is fed to `load_and_regrid.py`, but in the `load_and_regrid.py`, `level` is local variable that is not returned at the end of the function. 
3. So in the `mean_climate_driver.py`, level is still remain in [hPa] (even after `level = level * 100` in `load_and_regrid.py`)
4. `result_dict["Variable"]["level"] = level * 100  # hPa to Pa` is for storing the level info in [Pa] (that is in the data) into the output JSON file. 

But as you pointed, above #4 can be problematic if ds.plev.units != "Pa" because then it is adding false info. To keep it simple, I like the change you proposed in #1132.

**Comment by lee1043** [comment]
Closing the issue as it was resolved by merging PR #1132.
